Sources/Basic/OrderedSet.swift

```diff
@@ -13,7 +13,7 @@
 public struct OrderedSet<E: Hashable>: Equatable, Collection {
     public typealias Element = E
     public typealias Index = Int
-    public typealias Indices = CountableRange<Int>
+    public typealias Indices = Range<Int>
```

Comment:
We want to maintain compatibility in Basic/Utility targets with Swift (4.0.0). Can you fix this warning in a way that it doesn't break source compatibility? I think you should be able to use `#if swift(>=4.1)` or something similar. /cc @jrose-apple